llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -2652,6 +2657,9 @@ NVPTXTargetLowering::LowerOperation(SDValue Op, SelectionDAG &DAG) const {
   case ISD::SREM:
   case ISD::UREM:
     return LowerVectorArith(Op, DAG);
+  case ISD::DYNAMIC_STACKALLOC:
+    report_fatal_error(
```

Comment:
FWIW, using `report_fatal_error` will also generate stack dumps, which is misleading for a compiler bug.

The correctly way to do such thing is trigger a diagnostic, and let the frontend (e.g. clang) to report it.

See: https://github.com/llvm/llvm-project/blob/1d56138d741ebec81ac7a6dada10f5d6b891a768/llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp#L1291C1-L1300C2

Reply:
Thanks for your feedback!
I'll update the PR to trigger a diagnostic instead.

P.s.
I mimicked the `report_fatal_error` approach from BPFISelLowering.
That part could use a similar improvement as well.

https://github.com/llvm/llvm-project/blob/a2161154334355e9ecdf3cc3a0303b9481aad897/llvm/lib/Target/BPF/BPFISelLowering.cpp#L311 .